engines/tika/src/main/java/org/alfresco/transform/tika/transformers/Tika.java

```diff
@@ -0,0 +1,422 @@
+                targetMimetype = getValue(arg, true, targetMimetype, TARGET_MIMETYPE);
+            }
+            else if (arg.startsWith(NOT_EXTRACT_BOOKMARKS_TEXT))
+            {
+                getValue(arg, false, notExtractBookmarksText, NOT_EXTRACT_BOOKMARKS_TEXT);
+                notExtractBookmarksText = true;
+            }
+            else
+            {
+                throw new IllegalArgumentException("Unexpected argument " + arg);
+            }
+        }
+        includeContents = includeContents == null ? false : includeContents;
+        notExtractBookmarksText = notExtractBookmarksText == null ? false : notExtractBookmarksText;
+
+        transform(parser, documentSelector, includeContents, notExtractBookmarksText, inputStream,
+            outputStream, targetMimetype, targetEncoding);
+    }
+
+    private String getValue(String arg, boolean valueExpected, Object value, String optionName)
+    {
+        if (value != null)
+        {
+            throw new IllegalArgumentException("Duplicate " + optionName);
+        }
+        String stringValue = arg.substring(optionName.length()).trim();
+        if (!valueExpected && stringValue.length() > 0)
+        {
+            throw new IllegalArgumentException("Unexpected value with " + optionName);
+        }
+        if (valueExpected && stringValue.length() == 0)
+        {
+            throw new IllegalArgumentException("Expected value with " + optionName);
+        }
+        return stringValue;
+    }
+
+    private void transform(Parser parser, DocumentSelector documentSelector,
+        Boolean includeContents,
+        Boolean notExtractBookmarksText,
+        InputStream inputStream,
+        OutputStream outputStream, String targetMimetype, String targetEncoding)
+    {
+        try (Writer ow = new BufferedWriter(new OutputStreamWriter(outputStream, targetEncoding)))
+        {
+            Metadata metadata = new Metadata();
+            ParseContext context = buildParseContext(documentSelector, includeContents,
+                notExtractBookmarksText);
+            ContentHandler handler = getContentHandler(targetMimetype, ow);
+
+            parser.parse(inputStream, handler, metadata, context);
+        }
+        catch (UnsupportedEncodingException e)
+        {
+            throw new IllegalArgumentException("Unsupported encoding "+e.getMessage(), e);
+        }
+        catch (SAXException | TikaException | IOException e)
+        {
+            throw new IllegalStateException(e.getMessage(), e);
+        }
+    }
+
+    private ContentHandler getContentHandler(String targetMimetype, Writer output)
+    {
+        try
+        {
+            ContentHandler handler;
+            if (MIMETYPE_TEXT_PLAIN.equals(targetMimetype))
+            {
+                handler = new BodyContentHandler(output);
+            }
+            else
+            {
+                SAXTransformerFactory factory = (SAXTransformerFactory) SAXTransformerFactory.newInstance();
```

Comment:
Use static access with "javax.xml.transform.TransformerFactory" for "newInstance".